examples/extensions/add_prices.py

```diff
@@ -0,0 +1,171 @@
+                    'Hair Cuts',
+                    'Once a month',
+                    75000000,  # 75 USD
+                    'USD',
+                    client.get_type('PriceExtensionPriceUnitEnum').PER_MONTH,
+                    'http://www.example.com/haircuts',
+                    'http://m.example.com/haircuts'),
+        create_price_offer(client, 'Skin Care Package',
+                    'Four times a month',
+                    250000000,  # 250 USD
+                    'USD',
+                    client.get_type('PriceExtensionPriceUnitEnum').PER_MONTH,
+                    'http://www.example.com/skincarepackage')
+    ])
+
+    # Create a customer extension setting using the previously created
+    # extension feed item. This associates the price extension to your
+    # account.
+    extension_feed_item_operation = (
+        client.get_type('ExtensionFeedItemOperation', version='v2'))
+    extension_feed_item = extension_feed_item_operation.create
+    extension_feed_item.extension_type = (
+        client.get_type('ExtensionTypeEnum').PRICE)
+    extension_feed_item.price_feed_item.CopyFrom(price_feed_item)
+    extension_feed_item.targeted_campaign.value = (
+        CampaignServiceClient.campaign_path(customer_id, campaign_id))
+    extension_feed_item.ad_schedules.extend([
+        create_ad_schedule_info(client,
+            client.get_type('DayOfWeekEnum', version='v2').SUNDAY,
+            10,
+            client.get_type('MinuteOfHourEnum', version='v2').ZERO,
+            18,
+            client.get_type('MinuteOfHourEnum', version='v2').ZERO),
+        create_ad_schedule_info(client, 
+            client.get_type('DayOfWeekEnum', version='v2').SATURDAY,
+            10,
+            client.get_type('MinuteOfHourEnum', version='v2').ZERO,
+            22,
+            client.get_type('MinuteOfHourEnum', version='v2').ZERO)
+    ])
+
+    # Add the extension
+    try:
+        feed_service = client.get_service('ExtensionFeedItemService', version='v2')
+        # Issues a mutate request to add the customer extension setting and
+        # print its information.
+        feed_response = (
+            feed_service.mutate_extension_feed_items(customer_id, 
+                [extension_feed_item_operation])
+        )
+    except GoogleAdsException as ex:
+        print('Request with ID "%s" failed with status "%s" and includes the '
+              'following errors:' % (ex.request_id, ex.error.code().name))
+        for error in ex.failure.errors:
+            print('\tError with message "%s".' % error.message)
+            if error.location:
+                for field_path_element in error.location.field_path_elements:
+                    print('\t\tOn field: %s' % field_path_element.field_name)
+        sys.exit(1)
+
+    print('Created extension feed with resource name {}.'
+          .format(feed_response.results[0].resource_name))
+
+
+def create_price_offer(client, header, description, price_in_micros,
```

Comment:
You probably don't intend to share these helper methods. I'd recommend representing top-level methods or fields in a module as protected in that case.

Reply:
fixed